micrometer-observation/src/main/java/io/micrometer/observation/contextpropagation/ObservationThreadLocalAccessor.java

```diff
@@ -49,6 +50,7 @@ public class ObservationThreadLocalAccessor implements ThreadLocalAccessor<Obser
      */
     private ObservationRegistry observationRegistry = ObservationRegistry.create();
 
+    @SuppressWarnings("NullAway.Init")
```

Comment:
Shouldn't we mark this as `@Nullable`?

Reply:
It's a bit of an unusual setup, but I think the idea is that it's lazily initialized and non-null after initialization. We could defensively mark it as `@Nullable` but I thought this better expresses the intent.